geom/type_coordinates.go

```diff
@@ -18,3 +25,76 @@ type Coordinates struct {
 	// or not Z and M are populated.
 	Type CoordinatesType
 }
+
+// String gives a string representation of the coordinates.
+func (c Coordinates) String() string {
+	var sb strings.Builder
+	sb.WriteString("Coordinates[")
+	sb.WriteString(c.Type.String())
+	sb.WriteString("] ")
+	sb.WriteString(strconv.FormatFloat(c.X, 'f', -1, 64))
+	sb.WriteRune(' ')
+	sb.WriteString(strconv.FormatFloat(c.Y, 'f', -1, 64))
+	if c.Type.Is3D() {
+		sb.WriteRune(' ')
+		sb.WriteString(strconv.FormatFloat(c.Z, 'f', -1, 64))
+	}
+	if c.Type.IsMeasured() {
+		sb.WriteRune(' ')
+		sb.WriteString(strconv.FormatFloat(c.M, 'f', -1, 64))
+	}
+	return sb.String()
+}
+
+// appendFloat64s appends the coordinates to dst, taking into
+// consideration the coordinate type.
+func (c Coordinates) appendFloat64s(dst []float64) []float64 {
+	switch c.Type {
+	case DimXY:
+		return append(dst, c.X, c.Y)
+	case DimXYZ:
+		return append(dst, c.X, c.Y, c.Z)
+	case DimXYM:
+		return append(dst, c.X, c.Y, c.M)
+	case DimXYZM:
+		return append(dst, c.X, c.Y, c.Z, c.M)
+	default:
+		panic(c.Type.String())
+	}
+}
+
+// NewXYCoordinates constructs a new set of coordinates of type XY.
+func NewXYCoordinates(x, y float64) Coordinates {
```

Comment:
These probably aren't necessary anymore... I used then for tests before I switched from returning `[]Coordinates` to returning `Sequence`. Can be removed.

Reply:
Removed in 4f6f5b6